docs/config/@config.md

```diff
@@ -0,0 +1,197 @@
+|     category|`String` |  `"general"`  | Root element category, defaults to "general", if this is an empty string then the root category is disabled. Any primitive fields will cause an error, and you must specify sub-category objects. 
+
+### Example
+```java
+@Config(modid = "reallycoolmod")
+public class Configs {
+
+}
+```
+
+@Comment Use
+------------
+
+Adding a `@Comment` annotation to a field will add a comment to the config in it's file. It has 1 property:
+
+| Property|Type       | Default Value |Comment 
+|       -:|:-         |       :-:     |:-
+|    value|`String[]` |       N/A     |This can be passed as a `String` or `String[]`, a `String[]` will form a multi-line comment where 1 element = 1 line.
+
+### Example
+```java
+@Comment({
+  "You can add comments using this",
+  "and if you supply an array it will me multiline"
+})
+public static boolean doTheThing = true;
+```
+This will produce a `boolean` config value with name `doTheThing`, the defualt value `true` and the comment provided in the config file.
+
+@Name Use
+---------
+
+You should use the `@Name` annotation to give a config a user friendly name in the config file. This has 1 property:
+| Property | Type    | Default Value |
+|        -:|:-       |      :-:      |
+|     value|`String` |      N/A      | 
+
+### Example
+```java
+@Name("FE/T for the thing")
+public static int thingFE = 50; 
+```
+This will produce an `int` config value with the name `FE/T for the thing`, and the default value `50` in the config file.
+
+@RangeInt Use
+-------------
+
+You should use `@RangeInt` to limit an `int` config value. It has a couple of properties:
+
+| Property|Type |    Default Value    |
+|       -:|:-   |        :-:          |
+|      min|int  | `Integer.MIN_VALUE` |
+|      max|int  | `Integer.MAX_VALUE` |
+
+### Example
+```java
+@RangeInt(min = 0) 
+public static int thingFE = 50;
+```
+This will produce an `int` config value with the name `thingFE`, and the default value `50` in the config file; it will ensure the value stays between the bounds of `0` and `Integer.MAX_VALUE`.
+
+@RangeDouble Use
+----------------
+
+You should use `@RangeDouble` to limit a `double` config value. It has a couple of properties:
+
+| Property|Type   |    Default Value   |
+|       -:|:-     |        :-:         |
+|      min|double | `Double.MIN_VALUE` |
+|      max|double | `Double.MAX_VALUE` |
+
+### Example
+```java
+@RangeDouble(min = 0, max = Math.PI) 
+public static double chanceToDrop = 2;
+```
+This will produce a `double` config value with the name `chanceToDrop`, and the default value `Math.PI` in the config file; it will ensure the value stays between the bounds of `0` and `Math.PI`.
+
+@LangKey Use
+------------
+
+You should use the `@LangKey` annotation to give a config a user friendly via .lang files in the mod options menu. NOTE: This does not affect anything in the config file itself. This has 1 property:
```

Comment:
`You should use the `@LangKey` annotation to give a config a user friendly via .lang files in the mod options menu.` does not make sense